catalog/resource_sql_table.go

```diff
@@ -332,6 +332,35 @@ func (ti *SqlTableInfo) diff(oldti *SqlTableInfo) ([]string, error) {
 		statements = append(statements, fmt.Sprintf("ALTER %s %s SET TBLPROPERTIES (%s)", typestring, ti.SQLFullName(), ti.serializeProperties()))
 	}
 
+	// Since we marked `column_info` with `force_new`, this should only include the case where the size of the list is unchanged.
+	if !reflect.DeepEqual(ti.ColumnInfos, oldti.ColumnInfos) {
+		for i, ci := range ti.ColumnInfos {
+			oldCi := oldti.ColumnInfos[i]
+			if ci.Name != oldCi.Name {
+				println("name not equal!")
+				statements = append(statements, fmt.Sprintf("ALTER %s %s RENAME COLUMN %s to %s", typestring, ti.SQLFullName(), oldCi.Name, ci.Name))
```

Comment:
the column name needs to be wrapped inside backticks, in case it contains conflicting characters